### PR TITLE
fixed sidebar function again

### LINKED ISSUE #1575: a small bug in "sidebar" function
The small bug is :
Using a phone browser, when you are already on the "my files" page, you open the sidebar and click the "my files" button, nothing happened, the sidebar is still there and can't close itself because the webpage doesn't refresh. then, you will feel it's a bug. 
actually, that's a bug in Vue, the official website has two solutions.

### PR DESCRIPTION
**Pull request by niubility000**
fixed #1575
fixed #1576

Ramiresviana, this time, I change the route-link to button. and also use "replace" to solve the the problem of "back" button function.

### CODE REVIEW

**Comment by niubility000** [comment]
I change "replace" to "push". Replace only save one "back" button click and may confuse the user.

File: frontend/src/components/Sidebar.vue [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-      <router-link
-        class="action"
-        to="/files/"
-        :aria-label="$t('sidebar.myFiles')"
-        :title="$t('sidebar.myFiles')"
+      <button
+         class="action"
+         @click="toRoot"
+         :aria-label="$t('sidebar.myFiles')"
+         :title="$t('sidebar.myFiles')"
```
Comment (ramiresviana) [inline frontend/src/components/Sidebar.vue]:
```suggestion
        class="action"
        @click="toRoot"
        :aria-label="$t('sidebar.myFiles')"
        :title="$t('sidebar.myFiles')"
```

File: frontend/src/components/Sidebar.vue [hunk 2]
```diff
@@ -34,15 +34,15 @@
...
-        <router-link
-          class="action"
-          to="/settings"
-          :aria-label="$t('sidebar.settings')"
-          :title="$t('sidebar.settings')"
+        <button
+           class="action"
+           @click="toSettings"
+           :aria-label="$t('sidebar.settings')"
+           :title="$t('sidebar.settings')"
```
Comment (ramiresviana) [inline frontend/src/components/Sidebar.vue]:
```suggestion
          class="action"
          @click="toSettings"
          :aria-label="$t('sidebar.settings')"
          :title="$t('sidebar.settings')"
```

File: frontend/src/components/Sidebar.vue [hunk 3]
```diff
@@ -125,6 +125,14 @@ export default {
...
   },
   methods: {
+    toRoot() {
+      this.$router.push({ path: "/files/" });
+      this.$store.commit('closeHovers');
+    },
+    toSettings() {
+      this.$router.push({ path: "/settings" });
+      this.$store.commit('closeHovers');
+    },
```
Comment (ramiresviana) [inline frontend/src/components/Sidebar.vue]:
```suggestion
    toRoot() {
      this.$router.push({ path: "/files/" }, () => {});
      this.$store.commit("closeHovers");
    },
    toSettings() {
      this.$router.push({ path: "/settings" }, () => {});
      this.$store.commit("closeHovers");
    },
```

**Review by ramiresviana — CHANGES_REQUESTED** [review]

**Review by ramiresviana — APPROVED** [review]

**Pull request merged**